libs/ui-toolkit/src/components/icon/vega-icons/vega-icon.tsx

```diff
@@ -0,0 +1,45 @@
+import classNames from 'classnames';
+import type { VegaIconNames } from './vega-icon-record';
+import { VegaIconNameMap } from './vega-icon-record';
+
+export interface VegaIconProps {
+  name: VegaIconNames;
+  className?: string;
+  size?: 2 | 3 | 4 | 6 | 8 | 10 | 12 | 14 | 16;
+  ariaLabel?: string;
+}
+
+export const VegaIcon = ({
+  size = 4,
+  name,
+  className,
+  ariaLabel,
+}: VegaIconProps) => {
+  const effectiveClassName = classNames(
+    'inline-block',
+    'fill-current',
+    'align-text-bottom',
+    'fill-current',
+    'shrink-0',
+    {
+      'w-2 h-2': size === 2,
+      'w-3 h-3': size === 3,
+      'w-4 h-4': size === 4,
+      'w-6 h-6': size === 6,
+      'w-8 h-8': size === 8,
+      'w-10 h-10': size === 10,
+      'w-12 h-12': size === 12,
+      'w-14 h-14': size === 14,
+      'w-16 h-16': size === 16,
+    },
+    className
+  );
+  return (
+    <img
```

Comment:
suggestion (non blocking): This is a nice tidy up, but I'd much prefer to have a regular inline svg here for easier debugging. Perhaps look into the blueprint implementation and see how they set this up so we can get inline svg

Reply:
updated to use svgs